### PR TITLE
[KNOXPOD] how to connect to data templates

### PR DESCRIPTION
**Pull request by Aylr**
Changes proposed in this pull request:
- This PR is another and another and another revision on how to connect to data guides.

Previous Design Review notes:
- This addresses feedback in the knoxpod design review on 5/17 and 5/21

GE-149

### CODE REVIEW

File: docs/guides/connecting_to_your_data/database/postgres.md [hunk 1]
```diff
@@ -1,152 +1,111 @@
...
-</p>
-</details>
+For this guide we will use a `connection_string` like this:
 
+```                                                                            
+postgresql+psycopg2://<USERNAME>:<PASSWORD>@<HOST>:<PORT>/<DATABASE>
+```   
 
-### 3. Load the DataContext into memory
+### 3. Load your project's DataContext into memory
```
Comment (Shinnnyshinshin) [inline docs/guides/connecting_to_your_data/database/postgres.md]: <3

File: docs/guides/connecting_to_your_data/database/postgres.md [hunk 2]
```diff
@@ -1,152 +1,111 @@
...
+## Next Steps
 
+<NextSteps />
 
-### Here is the full script from which snippets were taken
+## Additional Notes
 
-```python file=../../../../integration/code/connecting_to_your_data/database/postgres.py#L1-L67
-```
+To view the full script [see it on GitHub](https://github.com/great-expectations/great_expectations/blob/knoxpod/integration/code/connecting_to_your_data/database/postgres.py)
```
Comment (Shinnnyshinshin) [inline docs/guides/connecting_to_your_data/database/postgres.md]: Love this as well

File: docs/guides/connecting_to_your_data/database/postgres.md [hunk 3]
```diff
@@ -1,152 +1,111 @@
...
-  - Add blurb about ActiveDataConnectors here
-:::
+:::warning TODO
+Using this method secrets may be stored in a `great_expectations.yml` file risking leakage via source control.
 
-:::warning
-  - Add test for yaml here
+**Ideas**
+- Use the convenience function the `datasource new` notebooks use?
+- Port said function to the data context itself? (probably!)
```
Comment (Shinnnyshinshin) [inline docs/guides/connecting_to_your_data/database/postgres.md]: Yes and yes, if possible

File: docs/guides/connecting_to_your_data/database/postgres.md [hunk 4]
```diff
@@ -1,152 +1,111 @@
...
 
-Next make sure you have the necessary packages for Great Expectations to connect to your postgres database.
+First, install the necessary dependencies for Great Expectations to connect to your postgres database.
 
 ```console
 pip install sqlalchemy psycopg2
 ```
 
-### 2. Determine how to add credentials to configuration
+### 2. Choose a credential handling method
```
Comment (Shinnnyshinshin) [inline docs/guides/connecting_to_your_data/database/postgres.md]: Nit-picky, but this title sounds a bit .. clunky for me.

**Review by Shinnnyshinshin — COMMENTED** [review]
Current impression for the `postgres` template is that I like it overall :) Even with most of the didactic content taken out, I still felt that the guide flowed nicely. Seeing it "in person" really drove that point home for me.  I liked the changes to the titles (other than the one nit-pick) and the link to full script on Github at the end. 

Regarding how we should handle the CLI workflow (for pandas and actually postgres is involved too), I've added an update to the JIRA task (GE-149).  My current feeling is that I'd like to push for parity between the CLI + all the datasource combinations by having  both the `InferredAssetDataConnector` and `RuntimeDataConnector` by default. We can talk more about this. 

PS. I also tried fixing a broken link in the docs that was making the build fail, but was only partially successful :/

File: docs/guides/connecting_to_your_data/database/postgres.md (hunk 4 above)
Comment (Aylr) [reply docs/guides/connecting_to_your_data/database/postgres.md]: Agreed!

File: docs/guides/connecting_to_your_data/components/where_to_run_code.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+Get an environment to run the code in this guide. Please choose an option below.
+
+<Tabs
+  defaultValue='cli'
+  values={[
```
Comment (Shinnnyshinshin) [inline docs/guides/connecting_to_your_data/components/where_to_run_code.md]:
```suggestion
<Tabs
  groupId="yaml-or-python"
  defaultValue='cli'
  values={[
```

Comment (Shinnnyshinshin) [reply docs/guides/connecting_to_your_data/components/where_to_run_code.md]: This way the `YAML` , and `python` section will stay the same for all tabs.  I think CLI can be its own thing with one additional sentence that I described in a separate comment

File: integration/code/connecting_to_your_data/filesystem/pandas_python_example.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        "class_name": "PandasExecutionEngine",
+    },
+    "data_connectors": {
+        "default_runtime_data_connector_name": {
+            "class_name": "RuntimeDataConnector",
+            "batch_identifiers": ["default_identifier_name"],
+        }
+    },
+}
+
```
Comment (Shinnnyshinshin) [inline integration/code/connecting_to_your_data/filesystem/pandas_python_example.py]: Will need to add the `InferredAssetDataConnector` here and in the `pandas_example.py` file too. What to do with the `base_directory` is an open question, unfortunately.  Can we discuss how to approach adding / testing the `InferredAssetDataConnector` for Pandas/Spark before merging this PR?

File: docs/guides/connecting_to_your_data/components/where_to_run_code.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  {label: 'python', value:'python'},
+  ]}>
+  <TabItem value="cli">
+
+If you use the Great Expectations CLI, run this command to automatically generate a pre-configured jupyter notebook:
+
+```console
+great_expectations --v3-api datasource new
+```
+
```
Comment (Shinnnyshinshin) [inline docs/guides/connecting_to_your_data/components/where_to_run_code.md]: Since `CLI` is its own tab here, I think one more sentence saying something like : "running the CLI command will open up a separate Notebook, and you can follow along in the YAML workflow below" would help users get oriented. Also with the added bonus that `YAML` is the default tab that is selected, the guide will read seamlessly :)

**Review by Shinnnyshinshin — CHANGES_REQUESTED** [review]

File: integration/fixtures/no_datasources/great_expectations/great_expectations.yml [hunk 8]
```diff
@@ -13,19 +13,8 @@ config_version: 3.0
...
-      module_name: great_expectations.execution_engine
-    module_name: great_expectations.datasource
-    data_connectors:
-      default_runtime_data_connector_name:
-        batch_identifiers:
-          - default_identifier_name
-        module_name: great_expectations.datasource.data_connector
-        class_name: RuntimeDataConnector
+datasources: {}
+
```
Comment (Aylr) [inline integration/fixtures/no_datasources/great_expectations/great_expectations.yml:17]: All our how to connect guides should start with no datasources, hence the fixture change and rename.

File: docs/guides/connecting_to_your_data/components/where_to_run_code.md [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+  groupId="yaml-or-python"
+  defaultValue='cli'
+  values={[
+  {label: 'CLI + filesystem', value:'cli'},
+  {label: 'No CLI + filesystem', value:'yaml'},
+  {label: 'No CLI + no filesystem', value:'python'},
+  ]}>
+  <TabItem value="cli">
+
+If you use the Great Expectations CLI, run this command to automatically generate a pre-configured jupyter notebook. Then you can follow along in the YAML-based workflow below:
```
Comment (Shinnnyshinshin) [inline docs/guides/connecting_to_your_data/components/where_to_run_code.md:16]: I think this will address any potential "mismatch" we have between the tabs here and the tabs below. Thanks!

File: docs/guides/connecting_to_your_data/filesystem/pandas.md [hunk 10]
```diff
@@ -1,136 +1,131 @@
...
-**TODO** insert link or have this be an include.
+```python file=../../../../integration/code/connecting_to_your_data/filesystem/pandas_yaml_example.py#L39-L45
+```
+Then load data into the `Validator`.
+```python file=../../../../integration/code/connecting_to_your_data/filesystem/pandas_yaml_example.py#L53-L59
+```
 
-Right now a Batch is just Data + Metadata that is recognized by GE. If you want to actually Validate your data with Expectations, you will need a Batch + ExpectationSuite.
-:::
+<Congratulations />
```
Comment (Shinnnyshinshin) [inline docs/guides/connecting_to_your_data/filesystem/pandas.md:114]: Pandas-like how-to-guides will support both a `RuntimeBatchRequest` with path (using a `RuntimeDataConnector`) and a `BatchRequest` with `data_asset_name` (using an `InferredAssetDataConnector`). This will be submitted as a separate PR as discussed during review

File: integration/code/connecting_to_your_data/filesystem/pandas_python_example.py [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+context.add_datasource(**datasource_config)
+
+batch_request = RuntimeBatchRequest(
+    datasource_name="taxi_datasource",
+    data_connector_name="default_runtime_data_connector_name",
+    data_asset_name="<YOUR_MEANGINFUL_NAME>",  # this can be anything that identifies this data_asset for you
+    runtime_parameters={"path": "<PATH_TO_YOUR_DATA_HERE>"},  # Add your path here.
+    batch_identifiers={"default_identifier_name": "something_something"},
+)
+
```
Comment (Shinnnyshinshin) [inline integration/code/connecting_to_your_data/filesystem/pandas_python_example.py:47]: To add `BatchRequest` here

File: integration/fixtures/no_datasources/great_expectations/great_expectations.yml (hunk 8 above)
Comment (Shinnnyshinshin) [reply integration/fixtures/no_datasources/great_expectations/great_expectations.yml:17]: <3

**Review by Shinnnyshinshin — APPROVED** [review]
LGTM
